src/app/Table.elm

```diff
@@ -1,29 +1,91 @@
-module Table exposing (view, init, Msg)
+module Table exposing (view, init, initWithPlayersAndCup)
+
+{-|
+
+   This module handles the table of players,
+   and everything relating to that.
+
+-}
 
 import Html exposing (..)
+import Html.Events exposing (..)
+import Types exposing (PlayerID)
+import TableMsg exposing (Msg(..))
+import Dice
 
 
-{-
+type alias PlayerName =
+    String
 
-   This module handles the view rendering for the table of players
-   and everything relating to that.
 
--}
+type alias DiceCount =
+    Int
+
 
+type alias TablePlayer =
+    { id : PlayerID
+    , name : PlayerName
+    , diceCount : DiceCount
+    }
 
-type Model
-    = Model
 
+type alias TablePlayers =
+    List TablePlayer
 
-type Msg
-    = Msg
+
+type alias Model =
+    { tablePlayers : TablePlayers
+    , myCup : Dice.Cup
+    }
 
 
 init : Model
 init =
-    Model
+    { tablePlayers = noPlayers
+    , myCup = Dice.emptyCup
+    }
+
+
+initWithPlayersAndCup : List ( Int, String, Int ) -> List Int -> Model
+initWithPlayersAndCup tuples cup =
+    let
+        tupleToPlayer ( id, name, diceCount ) =
+            { id = id, name = name, diceCount = diceCount }
+    in
+        { tablePlayers = List.map tupleToPlayer tuples
+        , myCup = Dice.cupFromIntList cup
+        }
+
+
+noPlayers : TablePlayers
+noPlayers =
+    []
+
+
+testPlayers : TablePlayers
+testPlayers =
+    [ TablePlayer 1 "Joe" 6, TablePlayer 2 "Garrett" 5 ]
+
+
+view : (Msg -> msg) -> Model -> Html msg
+view privateMsg { tablePlayers } =
+    div []
+        [ button [ onClick <| privateMsg CallLiar ] [ text "Liar!" ]
```

Comment:
Probably a bad example / test private message name, because calling a liar is a server message, not a local one.

Was just trying to think of something that *would* be local, and I was like... well, changing the name in the view might be local... but to do that we'd probably want the name to change locally, but also have a message send to the server to change it.

Can you compose messages, I wonder? Like... if we decide to give this module its own state which gets mounted in the main app, how would we both update the model *and* make sure the message for sending the server data happens as well?

Be interested to explore this, and find it out.